packages/teleport/src/services/agents/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { App } from 'teleport/services/apps';
+import { Database } from 'teleport/services/databases';
+import { Node } from 'teleport/services/nodes';
+import { Kube } from 'teleport/services/kube';
+import { Desktop } from 'teleport/services/desktops';
+
+export type AgentKinds = App | Database | Node | Kube | Desktop;
+
+export type AgentResponse = {
+  agents: AgentKinds[];
```

Comment:
Why use `AgentKinds` instead of generics? 

```
export type AgentsResponse<T> = {
    agents: T[];
    ....
}```

Reply:
good catch, i think this will be better, revised